### PR TITLE
Added .so files to doctest.py for testing C extension modules

### PR DESCRIPTION
**Pull request by lgpage**
#1498

### CODE REVIEW

**Comment by lgpage** [comment]
Also tested on a Cython template project:
https://github.com/lgpage/cython-template-project/tree/pytest_doctest
[Build Status](https://travis-ci.org/lgpage/cython-template-project)

File: CHANGELOG.rst [hunk 1]
```diff
@@ -7,6 +7,9 @@
 
 *
 
+* Added ``.so`` files to ``doctest.py`` for testing C extension modules
+  (`#1498`_)
+
```
Comment (The-Compiler) [inline CHANGELOG.rst]: You forgot to add a link at the bottom :wink:

File: _pytest/doctest.py [hunk 2]
```diff
@@ -145,7 +143,6 @@ def get_optionflags(parent):
 
 
 class DoctestTextfile(DoctestItem, pytest.Module):
-
```
Comment (The-Compiler) [inline _pytest/doctest.py]: Those whitespace changes are unrelated, and wrong according to [pep8](https://www.python.org/dev/peps/pep-0008/#blank-lines)

Comment (lgpage) [inline _pytest/doctest.py]: Hmm how did I miss reading that in pep8?
I probably read it and ignored it due to how strange the code looks with extra lines in vim with folding. My bad. Fixed.

**Comment by nicoddemus** [comment]
Thanks for the PR.

I have a few comments:

* I think we should think more carefully about this. For pytest to start to assume that all `.so` modules should be collected as doctest files seems dangerous to me, they might not be even a python extension after all, but just a standard `.so` file... could you check what happens when that's the case?

* It must also consider `.dll` and `.pyd` files on Windows.

* Perhaps this should be a separate option? 

* This change definitely should go into `features` branch. Unfortunately for that you have to close this PR, rebase into `features` and open an new PR because GitHub doesn't allow you to change the target branch after creating a PR.
